Packs/ExtraHop/Integrations/ExtrahopRevealXEventCollector/ExtrahopRevealXEventCollector.py

```diff
@@ -0,0 +1,471 @@
+        Get the access token from the ExtraHop API.
+
+        Args:
+            client_id: The Client ID to use for authentication.
+            client_secret: The Client Secret to use for authentication.
+
+        Returns:
+            tuple[str,int]: The token and its expiration time in seconds received from the API.
+        """
+        demisto.info("Generating new authentication token.")
+
+        req_headers = {
+            "cache-control": "no-cache",
+            "content-type": "application/x-www-form-urlencoded",
+        }
+        req_body = {
+            "grant_type": "client_credentials",
+            "client_id": client_id,
+            "client_secret": client_secret,
+        }
+        response = self._http_request(
+            method="POST",
+            url_suffix="/oauth2/token",
+            data=req_body,
+            headers=req_headers
+        )
+        token = response.get("access_token")
+        expires_in = response.get("expires_in")
+
+        return token, expires_in
+
+    def get_extrahop_version(self):
+        """Retrieve the ExtraHop version."""
+        self.set_headers()
+        return self._http_request(method="GET", url_suffix="/api/v1/extrahop/version")
+
+    def detections_list(self, body):
+        """Retrieve the detections from Reveal(X).
+        """
+        # Make sure we have a valid token
+        self.set_headers()
+        return self._http_request("POST", url_suffix="/api/v1/detections/search", json_data=body)
+
+
+""" HELPER FUNCTIONS """
+
+
+def trim_spaces_from_args(args: dict) -> dict:
+    """Trim spaces from values of the args dict.
+
+    Args:
+        args: Dict to trim spaces from.
+
+    Returns:
+        Arguments after trim spaces.
+    """
+    for key, val in args.items():
+        if isinstance(val, str):
+            args[key] = val.strip()
+
+    return args
+
+
+def prepare_list_detections_output(detections) -> str:
+    """Prepare human-readable output for list-detections command.
+
+    Args:
+        detections: List of detection response from the API.
+
+    Returns:
+        markdown string to be displayed in the war room.
+    """
+    hr_outputs = []
+    headers = ["Detection ID", "Risk Score", "Description", "Categories", "Status", "Resolution", "Start Time"]
+    for detection in detections:
+        hr_output = {
+            "Detection ID": detection.get("id"),
+            "Risk Score": detection.get("risk_score"),
+            "Description": detection.get("description"),
+            "Categories": detection.get("categories"),
+            "Status": detection.get("status"),
+            "Start Time": detection.get("start_time"),
+            "Mod Time": detection.get("mod_time"),
+            "Resolution": detection.get("resolution"),
+        }
+        hr_outputs.append(hr_output)
+
+    return tableToMarkdown(f"Found {len(hr_outputs)} Detection(s)", hr_outputs, headers=headers, removeNull=True)
+
+
+def validate_version(client, last_run) -> None:
```

Comment:
Why do need this function at all? 
You can just write in the documentation what version we support. I don't think we need this check